### PR TITLE
Include model parameter placeholders in site-packages

### PR DESCRIPTION
**Pull request by farhadrgh**
This is to address #499  
Let's not merge this now. This will result in clean pip uninstall, however, there might be other (potentially better) solutions to this.

### CODE REVIEW

**Comment by IgnacioJPickering** [comment]
So pip doesn't remove stuff inside /torchani/ when you call pip uninstall?

**Comment by IgnacioJPickering** [comment]
@farhadrgh Looks like what pip does is track all the stuff that was downloaded during installation and then remove those files, don't know if there is any way to add extra files to what pip tracks, but I couldn't find anything else. The issue if we do this is that the model files will actually not be removed at all when you do pip uninstall, but I guess there really is no workaround for that.

Also the other issue is that some files will be inside resources/ (ANI1x) and others will be in .local (ANI2x/ANI1ccx) which is messy.

**Comment by IgnacioJPickering** [comment]
We should let users know somehow that files have been downloaded which will require a manual delete after pip uninstall is run, and also as #499 says, a manual explicit download where a path is specified would be ideal

**Comment by zasdfgbnm** [comment]
How about:

when searching for parameters, search the following order:
- the path specified by the user (if given)
- resources inside TorchANI installation
- `.local/torchani`

And if not found, download the parameters and store it to the following path:
- if given, the path specified by the user, otherwise,
- `.local/torchani`

**Comment by farhadrgh** [comment]
I have been trying different ways to tackle this and avoid a user-specified path. The problem with the user-specified path is that the path cannot be recorded/stored and the user might end up downloading the parameters more than once. Plus we are trying to avoid the addition of new files (such as info that stores these custom path) to the install dir.

One clean solution is to trick pip into tracking some placeholder files. However, I am not sure yet if the `pip install` ignores these empty placeholders or not.

File: torchani/models.py [hunk 1]
```diff
@@ -82,27 +82,34 @@ def _parse_neurochem_resources(info_file_path):
...
+                files = glob.glob(os.path.join(resource_path, extracted_name, "resources", "*"))
+                for f in files:
+                    try:
+                        shutil.move(f, resource_path)
+                    except shutil.Error:
+                        pass
+                shutil.rmtree(os.path.join(resource_path, extracted_name))
+
+            else:
+                resource_path = local_dir
```
Comment (farhadrgh) [inline torchani/models.py]: Also, check out the changes here. There was also a bug that I fixed along the way!
It was still trying to `move` files if they exist in `local_dir` and as a result, were not downloaded.

**Comment by IgnacioJPickering** [comment]
@farhadrgh so you are adding a lot of empty files with the same names as the files that will be downloaded? we should definitely first test somehow that will work.

Why do you say that it is not possible to store the user specified path? why not save it in a file in ``./local/torchanirc``? something like that?

**Comment by farhadrgh** [comment]
Yes, these are placeholders for the parameter files and will be overwritten the first time use executes built-in functions. This is to avoid having files/path info saved in multiple locations. Of course not in case if permission error!

**Comment by IgnacioJPickering** [comment]
@farhadrgh If this works then it is probably the best solution I guess, how did you come up with this?

**Comment by farhadrgh** [comment]
> @farhadrgh If this works then it is probably the best solution I guess, how did you come up with this?

Literally spend some hours today banging my head to the wall! :D

**Comment by IgnacioJPickering** [comment]
@farhadrgh I think this is good but I don't know how to test it without deploying

**Comment by farhadrgh** [comment]
> @farhadrgh I think this is good but I don't know how to test it without deploying

Let's keep it WIP. I will check something tomorrow. Meanwhile, @zasdfgbnm may also have some opinion

**Comment by zasdfgbnm** [comment]
To test, can you generate the wheel and install the wheel?

**Comment by farhadrgh** [comment]
The wheel is generated successfully, the parameter files are removed with `pip uninstall` after being downloaded with model call.

To generate and install the wheel:

```bash
python setup.py bdist_wheel
pip install <path_to.whl extension>
```

**Pull request merged**